caluma/caluma_form/tests/test_form.py

```diff
@@ -63,6 +63,52 @@ def test_query_all_forms(
     snapshot.assert_match(result.data)
 
 
+def test_query_root_forms(
+    db,
+    snapshot,
+    form,
+    form_factory,
+    form_question,
+    form_question_factory,
+    question,
+    schema_executor,
+):
+    form_factory(name="3rd", description="Second result")
+    form_factory(name="2nd", description="Second result")
+    form_question_factory(form=form)
+
+    query = """
+        query GetRootForms($slug: String) {
+          allForms (
+            filter: [
+              {slug: $slug}
+              { metaValue: [{key: "is-main-form", value: true}]}
+            ]
+            order: [{attribute: NAME, direction: ASC}]
+          ){
+            edges {
+              node {
+                slug
+                name
+                meta
+                isPublished
+                source {
+                  slug
+                }
+              }
+            }
+          }
+        }
```

Comment:
Let's keep the test simple:

```suggestion
        query Form($slug: String) {
          allForms(filter: [{ slug: $slug }]) {
            totalCount
          }
        }
```